File: cpp/41c_509A_MaximumInTable.cpp
#include<bits/stdc++.h>

int main(){
    int n;
    std::cin>>n;
    std::vector<std::vector<int>> arr;
    for(int i = 0; i < n; i++){
        std::vector<int> nxt;
        for(int j = 0; j < n; j++){
            if (i == 0 || j == 0){
                nxt.push_back(1);
            }
            else{
                nxt.push_back(nxt[j-1] +arr[i-1][j]);
            }
        }
        arr.push_back(nxt);
    }
    std::cout<<arr[n-1][n-1];
    return 0;
}

File: python/41c_509A_MaximumInTable.py
def main():
    n = int(input())
    arr = []
    for i in range(n):
        nxt = []
        for j in range(n):
            if i == 0 or j == 0:
                nxt.append(1)
            else:
                nxt.append(nxt[-1]+arr[i-1][j])
        arr.append(nxt)
    print(arr[n-1][n-1])

if __name__ == "__main__":
    main()